ospd/ospd.py

```diff
@@ -584,23 +585,40 @@ def remove_scan_hosts_from_target_progress(
             scan_id, exc_hosts_list
         )
 
-    def set_scan_host_finished(self, scan_id: str, host: str) -> None:
+    def set_scan_host_finished(
+        self,
+        scan_id: str,
+        host: str = None,
+        finished_host_batch: List[str] = None,
+    ) -> None:
         """ Add the host in a list of finished hosts """
-        self.scan_collection.set_host_finished(scan_id, host)
+        if host:
+            finished_host_batch = [host]
+
+        self.scan_collection.set_host_finished(scan_id, finished_host_batch)
 
     def set_scan_progress(self, scan_id: str, progress: int) -> None:
         """ Sets scan_id scan's progress which is a number
         between 0 and 100. """
         self.scan_collection.set_progress(scan_id, progress)
 
     def set_scan_host_progress(
```

Comment:
For this function it is better to create a new method like
```python
def set_scan_progress_batch(self, scan_id: str, host_progress: Dict[str, int] = none)
```
and change `set_scan_host_progress` to call the new method set_scan_progress_batch internally